src/rest.cpp

```diff
@@ -269,6 +269,56 @@ static bool rest_block_notxdetails(HTTPRequest* req, const std::string& strURIPa
     return rest_block(req, strURIPart, false);
 }
 
+static bool rest_blockhash(HTTPRequest* req, const std::string& strURIPart)
+{
+    if (!CheckWarmup(req))
+        return false;
+    std::string strHeight;
+    const RetFormat rf = ParseDataFormat(strHeight, strURIPart);
+
+    int32_t nHeight;
+    if (!ParseInt32(strHeight, &nHeight))
+        return RESTERR(req, HTTP_BAD_REQUEST, "Parse error");
+
+    LOCK(cs_main);
+    if (nHeight < 0 || nHeight > chainActive.Height())
+        return RESTERR(req, HTTP_BAD_REQUEST, "Block height out of range: " + strHeight);
+
+    const CBlockIndex *pindex = chainActive[nHeight];
+
+    switch (rf) {
+    case RF_BINARY: {
+        CDataStream ssGetBlockHashResponse(SER_NETWORK, PROTOCOL_VERSION);
+        ssGetBlockHashResponse << pindex->GetBlockHash();
+        std::string binaryHash = ssGetBlockHashResponse.str();
+
+        req->WriteHeader("Content-Type", "application/octet-stream");
+        req->WriteReply(HTTP_OK, binaryHash);
+        return true;
+    }
+    case RF_HEX: {
+        CDataStream ssGetBlockHashResponse(SER_NETWORK, PROTOCOL_VERSION);
+        ssGetBlockHashResponse << pindex->GetBlockHash();
+        std::string strHex = HexStr(ssGetBlockHashResponse.begin(), ssGetBlockHashResponse.end()) + "\n";
+
+        req->WriteHeader("Content-Type", "text/plain");
+        req->WriteReply(HTTP_OK, strHex);
```

Comment:
Instead of forming the stream, use `req->WriteReply(HTTP_OK, pindex->GetBlockHash()->GetHex());`?

Reply:
Yeah I was wondering about this...I had just followed the pattern for the other endpoints' hex formats.  Directly writing the output of pindex->GetBlockHash().GetHex() reverses the order of the bytes as compared to HexStr().  I'm not sure which is correct?